shell/components/form/NameNsDescription.vue

```diff
@@ -169,106 +172,97 @@ export default {
   },
 
   data() {
-    return {
-      namespace:       '',
-      name:            '',
-      description:     '',
-      createNamespace: false,
-    };
+    return { createNamespace: false };
   },
 
-  created() {
-    const v = this.value;
-    const metadata = v.metadata;
-    let namespace, name, description;
-
-    if (this.nameKey) {
-      name = get(v, this.nameKey);
-    } else {
-      name = metadata?.name;
-    }
+  setup(props, { emit }) {
```

Comment:
May I recommend adding some code comments in a long code for the next time, considering the efforts to understand all the logic?
I have been struggling several times to code or review PRs for this very reason.

Kudos to you for refactoring this btw 🍺

Reply:
I'm open to adding comments where they will help. Let me know if you would like to sync on where we can benefit with additional comments

Reply:
Nah, let's keep it for the next time :)